### PR TITLE
[Stateful Stream Processing] Check for the presence of meroxa resources 

### PR DESCRIPTION
**Pull request by ericcheatham**
## Description of change
Executing a `.jar` in order to extract IR information from without first checking if meroxa resources exist in the `.jar` has the side effect of running the job it its entirety if the job does not have meroxa resources. This may happen twice:
- Once during upload when inspecting
- Once after its been uploaded to our cluster and ran as would be expected

This could be a problem in out-of-scope execution of jobs may result in non-determinant results on the consumer's end

Fixes https://github.com/meroxa/product/issues/945

## Type of change

- [ ]  New feature
- [x]  Bug fix
- [x]  Refactor
- [ ]  Documentation

## How was this tested?

- [ ]  Unit Tests
- [x]  Tested in staging
- [ ]  Tested in minikube

### CODE REVIEW

**Comment by janelletavares** [comment]
I don't think we need any of the doc updates right now, do we?

**Comment by ericcheatham** [comment]
Probably not but also this happens every single time i run a make build on this repo.

**Comment by samirketema** [comment]
@ericcheatham there's a env var `REBUILD_DOCS` in the makefile. It defaults to true, so that explains why it makes the changes when you do `make build`

**Comment by ericcheatham** [comment]
@samirketema  
image


Good catch.

File: cmd/meroxa/flink/deploy.go [hunk 1]
```diff
@@ -24,77 +26,105 @@ const (
 	irVal            = "EMIT_IR"
 )
 
+func doesJobUseMeroxaPlatform(ctx context.Context, jarPath string) (bool, error) {
```
Comment (samirketema) [inline cmd/meroxa/flink/deploy.go:29]: I think this approach is fine for now - do you want to leave a TODO comment (with a github issue) indicating that we need to verify that the `execute()` method is called on `MeroxaExecutionEnvironment`?

File: cmd/meroxa/flink/deploy.go [hunk 2]
```diff
@@ -24,77 +26,105 @@ const (
...
-		return nil, fmt.Errorf("%s\n%v", output, err)
-	}
 
-	_, err = os.Stat(irFilepath)
-	if err != nil {
-		// @TODO try the docker way because the jar is skinny
-		// Otherwise, there are no Meroxa* classes in this main class
-		return nil, nil
-	}
+	if usesMeroxa {
```
Comment (samirketema) [inline cmd/meroxa/flink/deploy.go:60]: thoughts on checking if `!usesMeroxa` and then returning early? that way the rest of this code is not nested in a condition?

We could also add a comment explaining why we only do `java -jar` when it has the bindings

**Review by samirketema — COMMENTED** [review]
looks good, some minor suggestions

File: cmd/meroxa/flink/deploy.go (hunk 2 above)
Comment (ericcheatham) [reply cmd/meroxa/flink/deploy.go:60]: That's a good idea. Will definitely make it easier to read as well. 👏🏽

**Review by samirketema — APPROVED** [review]

**Pull request merged**